### PR TITLE
Improve readme to better explain the application and it's use case

### PR DESCRIPTION
**Pull request by kylef**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,7 +4,17 @@
 
 [![Circle CI Status](https://img.shields.io/circleci/project/apiaryio/polls-app.svg)](https://circleci.com/gh/apiaryio/polls-app/tree/master)
 
-This is a Swift iOS client for the [Polls API](https://github.com/apiaryio/polls-api). A simple application allowing users to view polls and vote in them.
+This is a Swift iPhone and iPad client for the [Polls API](https://github.com/apiaryio/polls-api). A simple application allowing users to view polls and vote in them.
```
Comment (smizell) [inline README.md]: The last sentence is actually a fragment. I think you can combine the two with a dash of some sort.

File: README.md [hunk 2]
```diff
@@ -4,7 +4,17 @@
...
+
+Polls is an example application utilising the [Hyperdrive](https://github.com/the-hypermedia-project/Hyperdrive) API client. Hyperdrive will communicate to the [Polls API](http://github.com/apiaryio/polls-api) without any hard-coded knowledge about how the API is implemented, just an understanding of the semantics the API is using.
+
+The application does this by learning how the API works at run-time. This can
+be achived in one of two ways:
+
+- By downloading the [API specification](apiary.apib) in [API Blueprint](https://apiblueprint.org/) format.
+- Talking directly to an API that exposes it's features and transitions in the API, such as with the the [Siren](https://github.com/kevinswiber/siren) hypermedia content type.
```
Comment (danielgtaylor) [inline README.md]: Double `the` :wink:

**Comment by danielgtaylor** [comment]
Looks good aside from the two minor nits above!

**Comment by smizell** [comment]
Good to go :+1:

**Pull request merged**